src/Qir/Tests/QIR-static/qir-test-ouput.cpp

```diff
@@ -22,10 +21,8 @@ TEST_CASE("QIR: Out.Message", "[qir.Out][qir.Out.Message]")
         Microsoft::Quantum::OutputStream::ScopedRedirector qOStreamRedirector(outStrStream);
 
         // Log something (to the redirected output):
-        QirString qstr{std::string(testStr1)};
-        Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(&qstr);
-        qstr.str = testStr2;
-        Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(&qstr);
+        Microsoft__Quantum__Testing__QIR__Out__MessageTest__Interop(testStr1.c_str());
+        Microsoft__Quantum__Testing__QIR__Out__MessageTest__Interop(testStr2.c_str());
```

Comment:
Wow! Rather dramatic change.

Before this change the `QirString*` was passed to the `Microsoft__Quantum__Testing__QIR__Out__MessageTest__body()`, 
the `Microsoft__Quantum__Testing__QIR__Out__MessageTest__body()` was passing that pointer to the `__quantum__rt__message()` (in the generated .ll), and `__quantum__rt__message()` was handling that `QirString*`.

Now after this change I see that the `const char*` is passed instead. But in this PR I don't see any changes in `__quantum__rt__message()`. This makes me think that in the generated .ll the function `Microsoft__Quantum__Testing__QIR__Out__MessageTest__Interop()` creates the instance of `QirString` and passes the pointer to it to the `__quantum__rt__message()`. Which means that the Q# compiler has also changed.
But I don't see that this PR changes the compiler package version.

So for now I have an impression that this test after this change started behaving differently. If the test still _succeeds_ then I would investigate why.

Reply:
The interop functions that do the conversion from C types to QIR typles have been produced since the first version of the QIR was merged to main (see https://github.com/microsoft/qsharp-compiler/pull/889 for the original PR that added this to feature branch). We really should have been using these interop functions all along, but we were also waiting for multiple entrypoint support (added in https://github.com/microsoft/qsharp-compiler/pull/881). I am finally circling back to these call points so that we can follow our own guidance, which is to use the `__Interop` functions everywhere and not call into any `__body` functions.

Reply:
Can you please post here the body of the generated `Microsoft__Quantum__Testing__QIR__Out__MessageTest__Interop()`?

I assume that the following functions (called from `Microsoft__Quantum__Testing__QIR__Out__MessageTest__Interop()`) are still the same:
```ll
define dllexport void @__quantum__rt__message(%String* %.str) {
  %str = bitcast %String* %.str to %"struct.QirString"*
  call void @quantum__rt__message(%"struct.QirString"* %str)
  ret void
}
```
```c++
    void quantum__rt__message(QirString* qstr)   // NOLINT
    {
        Microsoft::Quantum::OutputStream::Get() << qstr->str << std::endl;
    }
```

Reply:
Yup, this is all the same IR that has been generated since March, we just haven't been using it to it's full extent. Here is what the interop version of MessageTest looks like:
```ll
define void @Microsoft__Quantum__Testing__QIR__Out__MessageTest__Interop(i8* %msg) #0 {
entry:
  %0 = call %String* @__quantum__rt__string_create(i8* %msg)
  call void @Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(%String* %0)
  call void @__quantum__rt__string_update_reference_count(%String* %0, i32 -1)
  ret void
}
```